### DISCUSSION TITLE
Snapping errors with Euler bend

### DISCUSSION THREAD

**Discussion by jonnyroth** [gdsfactory/gdsfactory#2826] [Q&A] [answered]
I'm running into issues with jagged polygon edges (1nm spaced points creating acute angles) that lead to drc issues, using gf.path.extrude(gf.path.euler(...)) with a cross_section that is defined with cross_section.rib_with_trenches.  I don't fully understand how these jagged edges come about, but I only see them after I move+rotate the cell (increments of 90deg only) containing the Euler bend.  I am not overriding the euler bend's default setting npoints=None.

My questions are:
(1) Is there a best practice to avoid this?
(2) If one sets a cross_section.Section's simplify parameter to 0.001, will no fidelity be lost if the mask shop is using a 1nm database unit, or is a finer setting safer if the cell is to be moved/rotated later?

My current solution is to use the below simplify function on the entire cell that is later moved/rotated, and it seems to fix the DRC error, though I don't know if I'm losing any mask fidelity.  I have ~100 micron effective radius Euler bends. Before simplification, the offending layer's polygon edge point density is about 1/ 130nm.  After simplification it's about 1 / 650nm.

Thanks for your attention - If a full example would be more helpful I can probably create one next week.
ver = GDSFactory 7.23.0

```python
@gf.cell
def simplify(component: gf.Component,tolerance=0.001, only_layers=None, copy_ports=True): #only_layers should be None or list of (layer,datatype)
    new_component=gf.Component(f'{component.name}_simplified')
    flat_component=component.flatten()
    for polygon in flat_component.polygons:
        shapely_polygon = shapely.geometry.Polygon(polygon.points)
        if (only_layers is None) or ((polygon.layer,polygon.datatype) in only_layers):
            shapely_polygon=shapely.simplify(shapely_polygon, tolerance=tolerance) #polygon gets simplified here
        simplified_polygon = shapely_polygon.exterior.coords
        new_component.add_polygon(simplified_polygon, layer=(polygon.layer,polygon.datatype))
        if copy_ports:
            new_component.ports=component.ports
    return new_component
```

**Comment by joamatab** [comment, marked as answer]
Hi Jonny,

Jagged edges typically result from oversampling the GDS points and the necessity of aligning points to a 1nm grid. You can find more information on this [here](https://gdsfactory.github.io/gdsfactory/notebooks/12_grid_snapping.html).

I’ve observed this issue when receiving designs from Lumerical or COMSOL as well. The best solution I've found is, as you mentioned, to simplify the polygons. I usually maintain a tolerance of roughly 10 to 20nm steps in the points. The mask preparation team at the foundry will resample your GDS points anyway, so the exact tolerance you use is not critically important.

Let me know your thoughts.

Best,
Joaquin

**Reply by jonnyroth** [reply]
Thanks Joaquin for your thoughts on this, and for the reference page on grid snapping errors.

Generally my concern about reducing the resolution is that it could somehow lead to more scattering, or create a periodic sidewall corrugation.  But running into these issues probably requires simplifying to a much coarser tolerance than ~1-2nm, for waveguides for NIR wavelengths, fabricated with optical lithography.

In the current tape-out I ran into some more DRC issues with imported GDS that was fractured into many polygons, but I was able to solve those with the combination of growing by 2nm / simplifying / shrinking by 2nm.

Best,
Jon
